src/UnitTests/LanguageServerClient/UnitTests.LanguageServerClient/PythonLanguageClient.cs

```diff
@@ -0,0 +1,273 @@
+                    client = new PythonLanguageClient(joinableTaskContext, clientContext);
+                    _languageClients.Add(client);
+                }
+            }
+
+            if (client != null) {
+                await broker.LoadAsync(new PythonLanguageClientMetadata(null, clientContext.ContentTypeName), client);
+            }
+        }
+
+        public static PythonLanguageClient FindLanguageClient(string contentTypeName) {
+            if (contentTypeName == null) {
+                throw new ArgumentNullException(nameof(contentTypeName));
+            }
+
+            lock (_languageClients) {
+                return _languageClients.SingleOrDefault(lc => lc.ContentTypeName == contentTypeName);
+            }
+        }
+
+
+        public static void DisposeLanguageClient(string contentTypeName) {
+            if (contentTypeName == null) {
+                throw new ArgumentNullException(nameof(contentTypeName));
+            }
+
+            PythonLanguageClient client = null;
+            lock (_languageClients) {
+                client = _languageClients.SingleOrDefault(lc => lc.ContentTypeName == contentTypeName);
+                if (client != null) {
+                    _languageClients.Remove(client);
+                    client.Stop();
+                    client.Dispose();
+                }
+            }
+        }
+
+        public string ContentTypeName => _clientContext.ContentTypeName;
+
+        public string Name => "Python Language Extension";
+
+        public IEnumerable<string> ConfigurationSections {
+            get {
+                // Called by Microsoft.VisualStudio.LanguageServer.Client.RemoteLanguageServiceBroker.UpdateClientsWithConfigurationSettingsAsync
+                // Used to send LS WorkspaceDidChangeConfiguration notification
+                yield return "python";
+            }
+        }
+
+        // called from Microsoft.VisualStudio.LanguageServer.Client.RemoteLanguageClientInstance.InitializeAsync
+        // which sets Capabilities, RootPath, ProcessId, and InitializationOptions (to this property value)
+        // initParam.Capabilities.TextDocument.Rename = new DynamicRegistrationSetting(false); ??
+        // 
+        // in vscode, the equivalent is in src/client/activation/languageserver/analysisoptions
+        public object InitializationOptions { get; private set; }
+
+        // TODO: what do we do with this?
+        public IEnumerable<string> FilesToWatch => null;
+
+        public object MiddleLayer { get; private set; }
+
+        public object CustomMessageTarget { get; private set; }
+
+        public bool IsInitialized { get; private set; }
+
+        public event AsyncEventHandler<EventArgs> StartAsync;
+
+#pragma warning disable CS0067
+        public event AsyncEventHandler<EventArgs> StopAsync;
+#pragma warning restore CS0067
+
+        public async Task<Connection> ActivateAsync(CancellationToken token) {
+            if (_server == null) {
```

Comment:
lots of this complex handshake in LanguageClient for VS is due to complexity VS has such as UI thread, having multiple clients broadcast, ordering between multiple clients on message handling, client aggregation and etc which doesn't exist in our case. so it might be much simpler if we just took some implementation code but not use VS code as they are, and simplify them?